plugins/action/configuration_item_batch.py

```diff
@@ -0,0 +1,98 @@
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import re
+
+from ansible.plugins.action import ActionBase
+from ansible.utils.vars import merge_hash
+
+from jinja2.nativetypes import NativeEnvironment
+
+MAPPING = {"string": str, "int": int, "float": float, "id": lambda x: x}
+
+
+def validate(name, args, required, typ):
+    """
+    Make sure that required values are not None and that if the value is
+    present, it is of the correct type.
+    """
+    value = args.get(name)
+    if required and value is None:
+        raise Exception("{0} is required argument".format(name))
+    if value is not None and not isinstance(value, typ):
+        raise Exception("{0} should be {1}".format(name, typ))
+
+
+class ActionModule(ActionBase):
+
+    _VALID_ARGS = frozenset(
+        ("instance", "sys_class_name", "id_column", "dataset", "map")
+    )
+
+    def run(self, _tmp=None, task_vars=None):
+        self._supports_check_mode = True
+        self._supports_async = True
+
+        result = super(ActionModule, self).run(task_vars=task_vars)
+
+        wrap_async = self._task.async_val and not self._connection.has_native_async
+
+        try:
+            self.validate_arguments(self._task.args)
+            args = dict(
+                self._task.args,
+                dataset=self.build_asset(
+                    self._task.args["map"], self._task.args["dataset"]
+                ),
+                map=dict(),
+            )
+
+            return merge_hash(
+                result,
+                self._execute_module(
+                    module_name="servicenow.itsm.configuration_item_batch",
+                    module_args=args,
+                    task_vars=task_vars,
+                    wrap_async=wrap_async,
+                ),
+            )
+
+        finally:
+            if not wrap_async:
+                self._remove_tmp_path(self._connection._shell.tmpdir)
+
+    @staticmethod
+    def validate_arguments(args):
+        # We only validate arguments that we use. We let the module
+        # validate the rest (like auth data).
+
+        # Next three string validations might seem strange at first, but there
+        # is a reason for this strangenes. On python 2, we should consider
+        # string to be instance of str or unicode. On python 3, strings are
+        # always instances of str. In order to avoid having a separate
+        # validate calls for python 2 and python 3, we always pass a pair of
+        # types that just happen to be the same on python 3.
+        validate("dataset", args, required=True, typ=list)
+        validate("map", args, required=True, typ=dict)
+
+    @staticmethod
+    def build_asset(mapping, dataset):
+        assets = [{} for i in range(len(dataset))]
+        env = NativeEnvironment()
+        regex = r"\| *(?P<filter>string|int|float) *$"
```

Comment:
Not the most crucial optimization, but it feels sloppy leaving this optimization opportunity out.

```suggestion
        final_filter_regex = re.compile(r"\| *(?P<filter>string|int|float) *$")
```